### PR TITLE
Fix test issue: catch async task exception; rm invalid validation

### PR DESCRIPTION
**Pull request by LiweiPeng**

### CODE REVIEW

File: src/test/outmdsd/MockServer.cc [hunk 1]
```diff
@@ -197,6 +197,30 @@ MockServer::RunEchoTask(
...
+    }
+    catch(const std::exception& ex) {
+        Log(std::string("Error: RunEchoTask() failed: ") + ex.what());
+
+        std::unique_lock<std::mutex> lck(m_connMutex);
+        m_connfdSet.erase(connfd);
+        lck.unlock();
+
+        shutdown(connfd, SHUT_RDWR);
+        close(connfd);
```
Comment (jasonzio) [inline src/test/outmdsd/MockServer.cc]: Should you re-throw the exception so that the master thread sees it, too?

**Comment by jasonzio** [comment]
:clock1:

File: src/test/outmdsd/MockServer.cc (hunk 1 above)
Comment (LiweiPeng) [inline src/test/outmdsd/MockServer.cc]: the master thread doesn't care/catch these exceptions. it just starts async tasks. and wait for them to finish before it close. the right place is to handle the exception at task level and log errors if failed. if a test client needs to continue talking with MockServer, client should re-initiate socket/connection().

---
In reply to: [96991364](https://github.com/Azure/fluentd-plugin-mdsd/pull/3#discussion_r96991364) [](ancestors = 96991364)

**Comment by LiweiPeng** [comment]
Thanks. The std::async behavior is well documented at C++ standard like http://en.cppreference.com/w/cpp/thread/async. Basically, the exception thrown in the task will be stored in the future object. the exception can be caught/retrown at future.get/wait.

**Pull request merged**